### ISSUE TITLE
Improved workflow: Speedup pre-commit, sync up max line length

### ISSUE THREAD

**Issue by ismail-codinglab** [intershop/intershop-pwa#573]
**Is your feature request related to a problem? If yes, please describe it.**

I noticed within some teams that certain formatting/rules aren't applied that are defined in the pre-commit process.
When talking to some of them I noticed that they find the pre-commit taking too long and simply go for `--no-verify` to speed up committing code, causing issues later on.

**Describe the desired solution.**

Remove the `jest` step to speedup pre-commit and make this part of the development sub branches CI. e.g. someone pushes code / creates PR = run tests

**Provide additional context.**

I also wonder why the gitlab config is part of this source code as it doesn't run on gitlab CI

And while at it you might perhaps also fix this:
https://github.com/intershop/intershop-pwa/blob/983ae82f0d609ed7c0a8fe6c477aff88ca485689/.prettierrc.json#L2
https://github.com/intershop/intershop-pwa/blob/983ae82f0d609ed7c0a8fe6c477aff88ca485689/tslint.json#L50

Either all 120 or all 140, i personally prefer 120, but as long as they're in sync it's fine by me.

**Comment by ismail-codinglab** [comment]
@jometzner any progress on this?

**Comment by jometzner** [comment]
We're using [husky ](https://github.com/typicode/husky/tree/main/docs) to inject our linting, jest-testing and prettier-running process into the pre-commit hook of git. Yes, that sounds quite heavy. But all that only runs if your git staging area contains files with certain filename extensions. There are two options for you, 

1. your colleagues are committing smaller change sets or 
2. you're going to change the husky configuration in your git repo. Look for `lint-staged` in the `package.json`

I doubt that we indeed have a majority for this, to remove the jest testing call from that pre-commit hook. Obviously I might be wrong on this. If we receive more feedback like this, we can make the change here in our repo. In the meantime I kindly ask you to go with the two options I outlined above.

**Comment by jometzner** [comment]
Your other question on the different line length settings: A `tslint` configuration consists of rules and options for those rules. If that legacy options array starts with a boolean false, the rule is switched off and is not part off the `npm run lint` process. So we could probably delete that line. We will do that with the switch from `tslint` to `eslint`

**Comment by jometzner** [comment]
I agree on the removal of the gitlab pipeline. That's a leftover from the early days.

**Comment by ismail-codinglab** [comment]
Thank you for your reply!

Well it's no longer a problem to many if their IDE [skips the pre-commit by ticking a simple box](https://i.redd.it/cwvbyqp5sl721.jpg) which they don't know why it is there.

Most developers use more tools on top of what is provided, increasing the pre-commit time. I wouldn't agree to go with these workarounds or keeping a heavy pre-commit then it is to implement a CI/CD process that will do this work in the background and can be shared between devs.

I would agree with you on linting. But the tests should be part of a CI/CD process, especially [when they are hanging](https://github.com/intershop/intershop-pwa/pull/484) from time to time. Local tests will miss out all the post-merge tests

@dhhyi Your opinion?

Basically i'm asking to remove `jest` from pre-commit to get rid of high commit load times

**Comment by dhhyi** [comment]
Hi @ismail-codinglab,

I agree with @jometzner on keeping the jest pre-commit hook as it is.
If you have trouble motivating your colleagues, you can modify it for your project. Either remove it altogether or just remove the `--findRelatedTests` argument from jest. The latter one increases the amount of tests quite significantly if you are modifying core files.
I would also encourage the people I work with to make smaller commits, that are well scoped and well defined. This helps with reviewing code and also helps when you are looking for bugs and want to find out what any previous change was originally intended to implement. That's the way we have done it here and we don't find that much resistance to this approach.

A complete test run should always be part of the CI anyway as some artifacts are loosely coupled and `findRelatedTests` won't find them in rare cases.

We didn't actually integrate the PR #484 you have been linking. Maybe working without `--findRelatedTests` but with `--runInBand` is a solution that works for you.

**Comment by ismail-codinglab** [comment]
Makes sense, well thanks @jometzner @dhhyi !
I will try push for those changes internally
